### PR TITLE
Convert ClusterMap gauge metrics from Boolean to Long

### PR DESCRIPTION
**Pull request by pnarayanan**
Following current convention and changing gauge metrics
to Long.

### CODE REVIEW

File: ambry-clustermap/src/main/java/com.github.ambry.clustermap/FixedBackoffResourceStatePolicy.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public void onHardDown() {
    */
   @Override
   public void onHardUp() {
-    logger.info("Marking resource {} as Hard up", resource);
+    logger.trace("Marking resource {} as Hard up", resource);
```
Comment (nsivabalan) [inline ambry-clustermap/src/main/java/com.github.ambry.clustermap/FixedBackoffResourceStatePolicy.java]: how often does this happen? Why changed to trace.

File: ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixClusterManagerTest.java [hunk 2]
```diff
@@ -281,47 +279,51 @@ public void metricsTest() throws Exception {
...
+    assertEquals(0L, getGaugeValue("partitionSealedCount"));
+    assertEquals(helixCluster.getDiskCapacity(), getGaugeValue("rawTotalCapacityBytes"));
+    assertEquals(0L, getGaugeValue("isMajorityReplicasDownForAnyPartition"));
+    assertEquals(0L,
+        getGaugeValue(helixCluster.getDownInstances().iterator().next().replace('_', '-') + "-DataNodeResourceState"));
+    assertEquals(1L,
+        getGaugeValue(helixCluster.getUpInstances().iterator().next().replace('_', '-') + "-DataNodeResourceState"));
     helixCluster.bringAllInstancesDown();
-    assertTrue(getGaugeValue("isMajorityReplicasDownForAnyPartition", Boolean.class));
+    assertEquals(1L, getGaugeValue("isMajorityReplicasDownForAnyPartition"));
```
Comment (nsivabalan) [inline ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixClusterManagerTest.java]: shouldn't this be 0 ?

Comment (nsivabalan) [inline ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixClusterManagerTest.java]: got it. it is expected to be 1

Comment (pnarayanan) [inline ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixClusterManagerTest.java]: Why do you think so? true is replaced with 1, false with 0.

File: ambry-clustermap/src/main/java/com.github.ambry.clustermap/FixedBackoffResourceStatePolicy.java (hunk 1 above)
Comment (pnarayanan) [inline ambry-clustermap/src/main/java/com.github.ambry.clustermap/FixedBackoffResourceStatePolicy.java]: This gets called once for every node when a live instance change trigger comes in, so it is better as trace.

Comment (nsivabalan) [inline ambry-clustermap/src/main/java/com.github.ambry.clustermap/FixedBackoffResourceStatePolicy.java]: got you.

**Comment by nsivabalan** [comment]
Looks good. Merging.

**Pull request merged**